tests/test_crud.py

```diff
@@ -127,62 +131,225 @@ def test_duplicated(self):
+                                     data=update_data)
+
+        # Log the request and response for debugging
+        print(f"Request data for update: {update_data}")
+        print(f"Response1 status code: {response1.status_code}")
+        print(f"Response1 content: {response1.text}")
+
+        self.assertFalse(response1.ok)
+
+        # connected
+        datapoint_basis_update = DatapointUpdate(
             entity_id=self.test_entity.id,
             entity_type=self.test_entity.type,
             attribute_name=self.test_entity.get_attribute_names().pop(),
         )
-        response2 = requests.request("PUT", settings.GATEWAY_URL + "/data/" + object_id,
-                                     headers=headers,
-                                     data=datapoint_basis_update.json()
-                                     )
+        response2 = requests.request("PUT", settings.GATEWAY_URL + "/data/" + object_id, headers=headers,
+                                     data=datapoint_basis_update.json())
         self.assertTrue(response2.ok)
 
         # check update result
         datapoint_basis2: Datapoint = datapoint_basis.copy()
         datapoint_basis2.entity_id = self.test_entity.id
         datapoint_basis2.entity_type = self.test_entity.type
         datapoint_basis2.attribute_name = self.test_entity.get_attribute_names().pop()
-        # datapoint_basis2.matchDatapoint = True
+        # datapoint_basis2.connected = True
         response = requests.request("GET", settings.GATEWAY_URL + "/data/" + object_id)
         self.assertEqual(
-            datapoint_basis2.dict().pop("matchDatapoint"),
-            json.loads(response.text).pop("matchDatapoint")
+            datapoint_basis2.dict().pop("connected"),
+            json.loads(response.text).pop("connected")
+        )
+
+    def test_match_datapoints(self):
+        headers = {
+            'Accept': 'application/json'
+        }
+
+        # create matched datapoint
+        datapoint = Datapoint(
+            **{
+                "topic": "topic/of/match",
+                "jsonpath": "$..data_match",
+                "connected": True,
+                "entity_id": "dp:001",
+                "entity_type": "Device",
+                "attribute_name": "temperature"
+            }
+        )
+        response = requests.request("POST", settings.GATEWAY_URL + "/data", headers=headers, data=datapoint.json())
+        object_id = response.json()["object_id"]
+        print(f"Created matched datapoint with object_id: {object_id}")
+        print(f"Response for matched datapoint creation: {response.json()}")
+        self.assertTrue(response.ok)
+
+        # verify match status
+        response = requests.request("GET", settings.GATEWAY_URL + "/data/" + object_id + "/status")
+        print(f"Match status response for matched datapoint: {response.json()}")
+        self.assertTrue(response.ok)
+        self.assertTrue(response.json())
```

Comment:
The other tests all pass. But here the validation fails for me. I got "False" instead of "True"

Reply:
Fixed it in updated test_crud.py